### PR TITLE
Fix TimeAggregationUnit Bug

### PR DESCRIPTION
**Pull request by chakrabortyr**
## Description
Fixes a bug introduced by #717. Reopened since I messed up rebasing.

## Motivation and Context
To make up for my earlier snafu.

## Tests performed
Added additional edge cases to ensure that proper date ranges are handled.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: open_xdmod/modules/xdmod/integration_tests/lib/Controllers/UsageExplorerTest.php [hunk 1]
```diff
@@ -39,37 +43,33 @@ public function testCorruptRequestData($input, $expectedMessage)
...
-            "group_by": "none",
-            "start_date": "2017-05-01",
-            "end_date": "2017-05-31",
-            "statistic": "job_count",
-            "operation": "get_charts",
-            "controller_module": "user_interface"
-        }
-EOF;
         $tests = array();
+        $view = array(
```
[the remark this answers is not in the document]
Comment (chakrabortyr) [reply open_xdmod/modules/xdmod/integration_tests/lib/Controllers/UsageExplorerTest.php]: For some bizarre reason it seems to run even before setUpBeforeClass (Github Issue [here](https://github.com/sebastianbergmann/phpunit/issues/1206)). Looks like the data provider is decoupled from the test class setup.

File: classes/DataWarehouse/Query/TimeAggregationUnit.php [hunk 2]
```diff
@@ -62,52 +65,32 @@ public function getUnitName()
...
+        $query = 'SELECT
+        COALESCE(MIN(' . $unit_id . '), -1) as minPeriodId,
+        COALESCE(MAX(' . $unit_id . '), -1) as maxPeriodId
+        FROM
+            ' . $this->getAggTablePrefix() . $unit . ' p
+        JOIN
+            modw.' . $unit . 's u ON u.id = p.' . $unit_id . '
+        WHERE
+            u.' . $unit . '_start <= ? AND
+            u.' . $unit . '_end > ?';
```
Comment (jpwhite4) [inline classes/DataWarehouse/Query/TimeAggregationUnit.php]: Please explain why this is a > rather than >=

**Comment by jpwhite4** [comment]
I note that this code change does impact the plots. Original code:

xdmod_number_of_jobs_submitted_2016-12-01_to_2017-01-01

Rudra Code:

xdmod_number_of_jobs_submitted_2016-12-01_to_2017-01-01-1
